core/stores/redis/redis.go

```diff
@@ -150,6 +150,46 @@ func NewScript(script string) *Script {
 	return red.NewScript(script)
 }
 
+// Publish is redis publish command implementation
+func (s *Redis) Publish(channel string, message any) (int64, error) {
+	return s.PublishCtx(context.Background(), channel, message)
+}
+
+// PublishCtx is redis publish command implementation.
+func (s *Redis) PublishCtx(ctx context.Context, channel string, message any) (val int64, err error) {
+	err = s.brk.DoWithAcceptable(func() error {
+		conn, err := getRedis(s)
+		if err != nil {
+			return err
+		}
+
+		val, err = conn.Publish(ctx, channel, message).Result()
+		return err
+	}, acceptable)
+
+	return
+}
+
+// Subscribe is redis subscribe command implementation.
+func (s *Redis) Subscribe(channel ...string) (*red.PubSub, error) {
+	return s.SubscribeCtx(context.Background(), channel...)
+}
+
+// SubscribeCtx is redis subscribe command implementation.
+func (s *Redis) SubscribeCtx(ctx context.Context, channel ...string) (val *red.PubSub, err error) {
```

Comment:
I think we might discuss more on this.

Current implementation doesn't export go-redis APIs. But this change exported `red.PubSub`.

Reply:
Hi @kevwan, updated, please help to check is there correct? thanks.